pkg/virt-controller/watch/vm.go

```diff
@@ -2898,38 +2908,53 @@ func (c *VMController) trimDoneVolumeRequests(vm *virtv1.VirtualMachine) {
 }
 
 // addRestartRequiredIfNeeded adds the restartRequired condition to the VM if any non-live-updatable field was changed
-func (c *VMController) addRestartRequiredIfNeeded(lastSeenVMSpec *virtv1.VirtualMachineSpec, vm *virtv1.VirtualMachine) (*virtv1.VirtualMachine, bool) {
+func (c *VMController) addRestartRequiredIfNeeded(lastSeenVMSpec *virtv1.VirtualMachineSpec, vm *virtv1.VirtualMachine) bool {
 	if lastSeenVMSpec == nil {
-		return vm, false
+		return false
+	}
+
+	// Expand any instance types and preferences associated with lastSeenVMSpec or the current VM before working out if things are live-updatable
+	currentVM := vm.DeepCopy()
+	if err := c.instancetypeMethods.ApplyToVM(currentVM); err != nil {
+		return false
```

Comment:
Were conflicts already checked at this point? IOW can we say that no conflicts can occur here anymore? Not sure about returning `false` on error.

Reply:
Yeah true, there's a chance that someone could modify the CR manually to cause this to conflict suddenly but I don't think we need to defend against that. I'll think about this some more and might drop the err checking here or add another function to the module that explicitly ignores conflicts?